packages/client/src/createTimeseriesProperty.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type Attachment,
+  TimeseriesDurationMapping,
+  type TimeSeriesPoint,
+  type TimeSeriesProperty,
+  type TimeSeriesQuery,
+} from "@osdk/client.api";
+import type {
+  AbsoluteTimeRange,
+  RelativeTime,
+  RelativeTimeRange,
+  StreamTimeSeriesPointsRequest,
+  TimeRange,
+} from "@osdk/internal.foundry";
+import { Ontologies, OntologiesV2 } from "@osdk/internal.foundry";
+import type { MinimalClient } from "./MinimalClientContext.js";
+import {
+  iterateReadableStream,
+  parseStreamedResponse,
+} from "./util/streamutils.js";
+
+export function createTimeseriesProperty<T extends number | string>(
+  client: MinimalClient,
+  objectApiName: string,
+  primaryKey: any,
+  propertyName: string,
+): TimeSeriesProperty<T> {
+  return {
+    async getFirstPoint() {
+      return OntologiesV2.OntologyObjectsV2.getFirstPoint(
+        client,
+        await client.ontologyRid,
+        objectApiName,
+        primaryKey,
+        propertyName,
+      ) as Promise<TimeSeriesPoint<T>>;
+    },
+    async getLastPoint() {
+      return OntologiesV2.OntologyObjectsV2.getLastPoint(
+        client,
+        await client.ontologyRid,
+        objectApiName,
+        primaryKey,
+        propertyName,
+      ) as Promise<TimeSeriesPoint<T>>;
+    },
+    async getAllPoints(query: TimeSeriesQuery) {
+      return getAllTimeSeriesPoints(
+        client,
+        objectApiName,
+        primaryKey,
+        propertyName,
+        query,
+      );
+    },
+
+    asyncIterPoints(query: TimeSeriesQuery) {
+      return iterateTimeSeriesPoints(
+        client,
+        objectApiName,
+        primaryKey,
+        propertyName,
+        query,
+      );
+    },
+  };
+}
+
+async function getAllTimeSeriesPoints<T extends string | number>(
```

Comment:
A few questions:
1.  Why isn't this function implemented by calling `iterateTimeSeriesPoints`? It feels duplicative and hard to maintain as two parallel code paths.
2. Should we be exposing a `getAll` here?

Reply:
1. Totally right, I'll do that
2. We had a getAll in legacy so I exposed for parity, I don't have enough data on usage here though to see if thats how people use it or not